robot/web/stream/state/state.go

```diff
@@ -0,0 +1,483 @@
+// Package state controls the source of the RTP packets being written to the stream's subscribers
```

Comment:
This is where the logic lives for detecting if `rtp_passthrough` is supported by the camera and upgrading / downgrading to / from rtp_passthrough & gostream `GetImage` as appropriate.

It is also responsible for detecting if the SubscribeRTP Subscription has terminated (which happens if there is an error internal to the module, such as the RTSP camera stream changing to H265, or a reconfigure, or the module crashing) and restarting either `rtp_passthrough` or gostream's `GetImage` code path as appropriate.

It is also responsible for reference counting the number of active peers (aka callers to viam-server AddStream - the number of calls to viam-server RemoveStream) and terminating rtp_passthrough subscriptions / the gostream `GetImage` pipeline if there are no subscribers (to conserver CPU resources).